### PR TITLE
Adds Sticks of Butter, Mayonnaise, and Mix function to grinder

### PR DESCRIPTION
**Pull request by TehZombehz**
:cl: TehZombehz
add: Sticks of butter and liquid mayonnaise can now be produced via a new 'mix' function on the grinder. Please eat responsibly.
/:cl:

butter mayo
A new 'mix' function on the grinder can be used to make sticks of butter and liquid mayonnaise. Sticks of butter require 15 units of milk, and egg yolk is made into mayonnaise in a 1-to-1 ratio. 
There are currently no uses for either in cooking recipes ~~aside from frying them and being a fat slob~~, but I plan to add some in the future.
I believe butter makes for the cornerstone of much cooking, and mayonnaise is great for many creamy recipes. I figure space could benefit from their addition. 
It's been a while since I've made a new PR and I'm a little wary about messing with the grinder, so please tell me of any erroneous work.

### CODE REVIEW

**Comment by InsaneHyena** [comment]
I am Ivan Ivanovich, and I approve of this PR.

**Comment by Pascal125** [comment]
10/10 Merge this right now.

**Comment by kevinz000** [comment]
Yay :D

**Comment by Repukan** [comment]
Cyka blyat drink de fuggin mayonez

**Comment by Tacolizard** [comment]
_**d E e P f R i E d B u T t E r**_

**Comment by Anonmare** [comment]
To add: Adidas tracksuit, squatting and bandit radio

**Comment by Armhulen** [comment]
fuck I was gonna do the deep fried butter joke

**Comment by FantasticFwoosh** [comment]
Don't forget slavic milkmaids for churning like that live butter churning demo they did at eurovision.

**Comment by Supermichael777** [comment]
remove mayo worst spread

**Comment by bgobandit** [comment]
I approve of bandit radio

**Comment by PopNotes** [comment]
You want _another_ butterbar..?

**Comment by Armhulen** [comment]
ROD + BUTTER = BUTTER ON A STICK

then you deepfry it

**Comment by Armhulen** [comment]
make that also stop your heart

**Comment by Razharas** [comment]
mayo needs oil, lemon juice and mustard besides the eggs

**Comment by TehZombehz** [comment]
Throwing eggs in a machine and getting mayo works for games like harvest moon, it'll work for spess.

**Comment by Razharas** [comment]
We are above games which simplify their gameplay to appease filthy casuals

**Comment by TehZombehz** [comment]
Should I rework all cake to require butter?

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 1]
```diff
@@ -467,3 +472,32 @@
...
+		//For butter and other things that would change upon shaking or mixing
+		power_change()
+		if(stat & (NOPOWER|BROKEN))
+				return
+		if (!beaker)
+				return
+		playsound(src.loc, 'sound/machines/juicer.ogg', 20, 1)
+		var/offset = prob(50) ? -2 : 2
+		animate(src, pixel_x = pixel_x + offset, time = 0.2, loop = 250) //start shaking
+		operating = 1
```
Comment (lzimann) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: `TRUE`

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 2]
```diff
@@ -467,3 +472,32 @@
...
+		updateUsrDialog()
+		spawn(50)
```
Comment (lzimann) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: Use `addtimer` here, spawns are bad!

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 3]
```diff
@@ -467,3 +472,32 @@
...
+				pixel_x = initial(pixel_x) //return to its spot after shaking
+				operating = 0
```
Comment (lzimann) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: `FALSE`

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 4]
```diff
@@ -467,3 +472,32 @@
...
+				updateUsrDialog()
+				if (beaker.reagents.total_volume)
+						//Recipe to make Butter
+						if (beaker.reagents.has_reagent("milk", 15))
```
Comment (lzimann) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: You can just do
```
while(beaker.reagents.remove_reagent("milk", 15)
    new /obj/item/weapon/reagent_containers/food/snacks/butter(get_turf(src))
```

**Review by lzimann — CHANGES_REQUESTED** [review]
Some small things.

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm (hunk 4 above)
Comment (TehZombehz) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: This is the only part of this that is actually my code. The rest of this is changes you're asking me to make to pre-existing grinder code.

Comment (Iamgoofball) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: whoa man reagent code isnt set up for while()s @lzimann

Comment (lzimann) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm]: Ugh, why nothing works like I assume they do?

**Review by lzimann — APPROVED** [review]

**Pull request merged**